src/Pages/ProductListing/ProductListing.js

```diff
@@ -1,7 +1,103 @@
-import React from 'react'
+import React, {useState, useEffect} from 'react';
+import ProductCard from "../../Components/ProductCard/ProductCard";
+import axios from 'axios';
+import "./ProductListing.css";
 
 export default function ProductListing() {
+
+    const [products, setProducts] = useState([]);
+    const [categories, setCategories] = useState([]);
+  
+    useEffect(() => {
+      (async function getProducts() {
+        try {
+          const { data } = await axios.get("/api/products");
+          setProducts(data.products);
+        }
+       catch(error) {
+          console.log("Couldn't get data. "+error);
+       }
+       
+      })();
+    }, []);
+  
+    useEffect(() => {
+      (async function getCategories() {
+        try {
+          const { data } = await axios.get("/api/categories");
+          setCategories(data.categories);
+        }
+        catch(error) {
+          console.log("Couldn't get data. "+error);
+       }
+       
+      })();
+    }, []);
+
   return (
-    <div>ProductListing</div>
+    <div>
+
+<div className="productsContainer">
+        <aside class="filters">
+                                                <div class="filter-Clear">
+                                                        <span class="filter-heading">Filters</span>
+                                                        <span class="clearAll">Clear</span>
+                                                </div>
+                                                <div class="filter-Price">
+                                                        <div class="filter-heading price-label">Price</div>
+                                                        <input type="range" min="300" max="1000" value="100"
+                                                                class="slider" id="filterRange"/>
+                                                </div>
+                                                <div class="filter-Category">
+
+                                                        <div class="filter-heading">Category</div>
```

Comment:
Use Formatting.

Reply:
I was using prettier earlier but it was breaking my code. I will format all the pages in the end, manually or by using some other way. If you can suggest any way, it would be nice.

Reply:
Prettier works for me, but if it's breaking your code, you can use "beautify" or any other formatting tools.

Reply:
okay, thanks!